src/components/PhotoFrame.tsx

```diff
@@ -271,7 +276,8 @@ const PhotoFrame = ({
             if (galleryContext.thumbs.has(item.id)) {
                 url = galleryContext.thumbs.get(item.id);
             } else {
-                url = await DownloadManager.getPreview(item);
+                const response = DownloadManager.queueUpGetPreviewRequest(item);
```

Comment:
was thinking of naming them getPreview and internally calling this queueUpAndWait function, the gallery doesn't need to know that it is getting queued